### PR TITLE
Don't warn WKT imports

### PR DESCRIPTION
**Pull request by unmultimedio**

### CODE REVIEW

File: private/buf/cmd/buf/buf_test.go [hunk 1]
```diff
@@ -2419,7 +2419,19 @@ func testRunStdout(t *testing.T, stdin io.Reader, expectedExitCode int, expected
...
+		func(use string) *appcmd.Command { return NewRootCommand(use) },
+		expectedExitCode,
+		expectedStderr,
+		internaltesting.NewEnvFunc(t),
+		stdin,
+		args...,
+	)
+}
+
+func testRunStdoutStderrNoWarn(t *testing.T, stdin io.Reader, expectedExitCode int, expectedStdout string, expectedStderr string, args ...string) {
```
Comment (unmultimedio) [inline private/buf/cmd/buf/buf_test.go:2434]: This func doesn't output warnings. It sends `--no-warn` flag.

File: private/bufpkg/bufimage/bufimagebuild/builder.go [hunk 2]
```diff
@@ -268,6 +272,32 @@ func (b *builder) warnInvalidImports(
...
 }
 
+var (
+	wktFiles map[string]struct{} // populated in isWKTFile func
+	wktOnce  sync.Once
+)
+
+func isWKTFile(filepath string) bool {
+	wktOnce.Do(func() {
+		wktFiles = map[string]struct{}{
```
Comment (saquibmian) [inline private/bufpkg/bufimage/bufimagebuild/builder.go:282]: We actually have this data already, it's just unexported in `datawkt`. I wonder if we could change `wkt-go-data` to create a new exported `FilePaths []string` variable that contains this.

File: private/bufpkg/bufimage/bufimagebuild/builder.go [hunk 3]
```diff
@@ -268,6 +272,32 @@ func (b *builder) warnInvalidImports(
 	return nil
 }
 
+var (
```
Comment (bufdev) [inline private/bufpkg/bufimage/bufimagebuild/builder.go:275]: global vars always go at the top of a file (but in this case, they will be deleted)

File: private/bufpkg/bufimage/bufimagebuild/builder.go (hunk 2 above)
Comment (bufdev) [reply private/bufpkg/bufimage/bufimagebuild/builder.go:282]: https://github.com/bufbuild/buf/blob/main/private/gen/data/datawkt/datawkt.gen.go#L158

**Review by bufdev — CHANGES_REQUESTED** [review]

File: private/bufpkg/bufimage/bufimagebuild/builder.go (hunk 2 above)
Comment (saquibmian) [reply private/bufpkg/bufimage/bufimagebuild/builder.go:282]: Actually yeah, that's basically exactly this function.

File: private/buf/cmd/buf/buf_test.go (hunk 1 above)
Comment (bufdev) [reply private/buf/cmd/buf/buf_test.go:2434]: when did we add a `--no-warn` flag? is this part of the buf CLI?

File: private/bufpkg/bufimage/bufimagebuild/builder.go (hunk 2 above)
Comment (bufdev) [reply private/bufpkg/bufimage/bufimagebuild/builder.go:282]: it in fact IS exactly this function :-)

**Review by saquibmian — APPROVED** [review]

File: private/buf/cmd/buf/buf_test.go (hunk 1 above)
Comment (bufdev) [reply private/buf/cmd/buf/buf_test.go:2434]: oh got it, ok

**Review by bufdev — APPROVED** [review]

**Pull request merged**